src/components/screens/vacation-requests-screen.tsx

```diff
@@ -73,6 +80,57 @@ const VacationRequestsScreen = () => {
     fetchVacationRequestStatuses();
   }, [vacationRequests]);
 
+    /**
+   * Initialize logged in person data.
+   */
+  const getPersons = async () => {
+    if (persons.length) {
+      setLoading(true);
+      if (loggedInPerson || config.person.id) setLoading(false);
+      {
+        const fetchedPersons = await personsApi.listPersons({ active: true });
+        setPersons(fetchedPersons);
+      }
+    }
+  };
+
+  useMemo(() => {
+    getPersons();
+  }, [persons]);
+
+  /**
+   * Display persons vacation days
+   * @param person representing the individual's vacation details.
+   */
+  const renderVacationDays = (person: Person | undefined) => {
+    const spentVacationsColor =
+      person && person.spentVacations > 0 ? theme.palette.success.main : theme.palette.error.main;
+
+    const unspentVacationsColor =
+      person && person.unspentVacations > 0 ? theme.palette.success.main : theme.palette.error.main;
+
+    if (!person && !loading && persons.length) {
+      return <Typography>{strings.error.fetchFailedNoEntriesGeneral}</Typography>;
+    } else if (person) {
+      return (
+        <Box sx={{ display: "flex", justifyContent: "center" }}>
+          <Box sx={{ width: "40%" }}>
+            <Typography>
+              {strings.vacationsCard.spentVacations}
+              <span style={{ color: spentVacationsColor }}>{person.spentVacations}</span>
+            </Typography>
+          </Box>
+          <Box>
+            <Typography>
+              {strings.vacationsCard.unspentVacations}
+              <span style={{ color: unspentVacationsColor }}>{person.unspentVacations}</span>
+            </Typography>
+          </Box>
+        </Box>
+      );
+    }
+  };
```

Comment:
I agree. This could be made into it's own component as the only differences are related to the style of the returned parent component, the flex box. Remove the sx property from line 117 and overwrite component on line 116 to a React fragment component (which is this <></>), and then the imported component `renderVacationDays()` could be used like this, for example:

In vacation-requests-screen:
`<Box sx={{ display: "flex", justifyContent: "space-around" }}>renderVacationDays(...)</Box>`

In vacations-card:
`<Box sx={{ display: "flex", flexDirection: "column" }}>renderVacationDays(...)</Box>`

Reply:
Created new util, vacation-days-utils.